src/hwProvider.ts

```diff
@@ -31,47 +31,123 @@ export class HWProvider {
+            return this._transport;
+        }
+
+        const isLedgerSupported = await this.isLedgerTransportSupported();
 
-        if (webUSBSupported) {
-            return await TransportWebUSB.create();
+        if (!isLedgerSupported) {
+            throw Error("Ledger is not supported");
         }
 
-        let webHIDSupported = await TransportWebHID.isSupported();
-        if (webHIDSupported) {
-            return await TransportWebHID.open("");
+        try {
+            const webUSBSupported = await this.isWebUSBSupported();
+
+            if (webUSBSupported) {
+                return await TransportWebUSB.create();
+            }
+        } catch (error) {
+            console.error("Failed to create USB transport:", error);
+        }
+
+        try {
+            const webBLESupported = await this.isBLESupported();
+
+            if (webBLESupported) {
+               return await TransportWebBLE.create();
+            }
+        } catch (error) {
+            console.error("Failed to create BLE transport:", error);
         }
 
-        return await TransportU2f.create();
+        try {
+            const webHIDSupported = await this.isWebHIDSupported();
+
+            if (webHIDSupported) {
+                 return await TransportWebHID.create();
+            }
+        } catch (error) {
+            console.error("Failed to create HID transport:", error);
+        }
+
+        // Deprecated: Use WebHID instead
+        // https://developers.ledger.com/docs/transport/choose-the-transport/
+        // try {
+        //     return await TransportU2f.create();
+        // } catch (error) {
+        //     console.error("Failed to create HID transport:", error);
+        // }
+
+        throw Error('Failed to initialize provider');
+    }
+
+    async isLedgerTransportSupported(): Promise<boolean> {
+        return await this.isBLESupported() || await this.isWebUSBSupported() || await this.isWebHIDSupported();
+
+    }
+
+    async isBLESupported(): Promise<boolean> {
+        return await TransportWebBLE.isSupported();
+    }
+
+    async isWebUSBSupported(): Promise<boolean> {
+        return await TransportWebUSB.isSupported() && platform.name !== "Opera";
+    }
+
+    async isWebHIDSupported(): Promise<boolean> {
+        return await TransportWebHID.isSupported();
     }
 
     /**
      * Returns true if init() was previously called successfully
      */
     isInitialized(): boolean {
-        return !!this.hwApp;
+        return Boolean(this.hwApp && this._transport);
     }
 
     /**
      * Mocked function, returns isInitialized as an async function
      */
     isConnected(): Promise<boolean> {
-        return new Promise((resolve, _) => resolve(this.isInitialized()));
+        return new Promise((resolve) => {
+            if (!this.isInitialized()) {
+                return resolve(false);
+            }
+
+            const bluetoothTransport = this._transport as TransportWebBLE;
```

Comment:
I am not sure if `as` behaves as intended in this case (sorry if I misunderstood).

Perhaps, instead, we should retain the transport "kind" upon initialization, then have a switch over here, in `isConnected()` (to perform the connection checks with respect to the transport "kind")?